File: Software/Machine_Learning_Python/utils_py/error_objects.py
"""
Author: Alon Haviv, Stellar Intelligence.

A module for error handler classes.
"""

class BaseException(Exception):
    """This base class represents a general error. Other error-classes will extend it."""

    def __init__(self, code: int, message: str=''):
        """
        Creates a new error object.
        code - A code from Consts.json.
        message - An optional message. Default value is: ''.
        """
        self.code = code
        self.message = message

    def __str__(self) -> str:
        return f'{self.__class__.__name__} Error code: {self.code}: {self.message}'

    def __repr__(self) -> str:
        return f'{self.__class__.__name__} Error code: {self.code}: {self.message}'

# Expose
__all__ = ['BaseException']

File: Software/Music_Transcription_App/utils/error_objects.js
/**
 * Author: Alon Haviv, Stellar Intelligence.
 * 
 * A module for error handler classes.
 * */

/**
 * This base class represents a general error. Other error-classes will extend it.
 * */
class BaseError {
    /**
     * A constructor.
     * @param {number} code - A code from Consts.json.
     * @param {string} message - An optional message. Default value is: ''.
     */
    constructor(code, message = '') {
        this.code = code;
        this.message = message;
    }

    /**
     * Implementing the toString() method.
     * */
    toString() {
        return `${this.constructor.name} Error code: ${this.code}: ${this.message}`
    }
}

/**
 * This class represents an error that describes a spawned Python process failure.
 * */
class SpawnProcessError extends BaseError {
    /**
     * A constructor.
     * @param {number} code - A code from Consts.json.
     * @param {string} message - An optional message. Default value is: ''.
     */
    constructor(code, message = '') {
        super(code, message);
    }
}

/**
 * This class represents an error that describes system operations failure (s.a. saving files).
 * */
class SystemError extends BaseError {
    /**
     * A constructor.
     * @param {number} code - A code from Consts.json.
     * @param {string} message - An optional message. Default value is: ''.
     */
    constructor(code, message = '') {
        super(code, message);
    }
}

// Export the classes:
module.exports = {
    BaseError: BaseError,
    SpawnProcessError: SpawnProcessError,
    SystemError: SystemError
};